src/discordtipbot/middleware/errorcatcher.cr

```diff
@@ -0,0 +1,7 @@
+class ErrorCatcher
+  def call(payload, context)
+    yield
+  rescue ex
+    Raven.capture(Exception.new("Exception while handling message #{payload}", ex))
```

Comment:
Don't forget, you can be pretty creative with these, like letting the user know why their command was ignored, or didn't complete:

```cr
rescue ex
  ctx[Discord::Client].create_message(payload.channel_id, "Sorry, something unexpected went wrong!")
  Raven.capture(Exception.new("Exception while handling message #{payload}", ex))
```